README.md

```diff
@@ -82,16 +107,52 @@ amending `options` according to your back end configuration:
 <script type="module">
     import * as webeid from 'web-eid.js';
 
+    const lang = navigator.language.substr(0, 2);
     const signButton = document.querySelector("#webeid-sign-button");
 
     signButton.addEventListener("click", async () => {
-        const options = {
-            postPrepareSigningUrl: window.location.origin + "/sign/prepare",
-            postFinalizeSigningUrl: window.location.origin + "/sign/sign"
-        };
-
         try {
-            const response = await webeid.sign(options);
+            const {
+                certificate,
+                supportedSignatureAlgorithms,
+            } = await webeid.getSigningCertificate({lang});
+            
+            const prepareSigningResponse = await fetch("/sign/prepare", {
+                method: "POST",
+                headers: {
+                    "Content-Type": "application/json",
+                    [csrfHeaderName]: csrfToken
+                },
+                body: JSON.stringify({certificate, supportedSignatureAlgorithms}),
+            });
+            if (!prepareSigningResponse.ok) {
+                throw new Error("POST /sign/prepare server error: " +
+                                prepareSigningResponse.status);
+            }
+            const {
+                hash,
+                hashFunction
+            } = await prepareSigningResponse.json();
+
+            const {
+                signatureAlgorithm,
+                signature,
+            } = await webeid.sign(certificate, hash, hashFunction, {lang});
+            
+            const finalizeSigningResponse = await fetch("/sign/sign", {
```

Comment:
Not a fan of repeating words in paths. Maybe `/sign/finalize` instead?
```suggestion
            const finalizeSigningResponse = await fetch("/sign/finalize", {
```

Or maybe `POST /document/12345/sign/prepare` and `POST /document/12345/sign`.  
I do understand what you mean by `/sign/sign`, but it might be a bit confusing for the users.

Reply:
Roger that, `"/sign/finalize"` it is then.